src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs

```diff
@@ -44,7 +47,7 @@ public ManagedIdentityClientAssertion(string? managedIdentityClientId)
         protected override async Task<ClientAssertion> GetClientAssertion(CancellationToken cancellationToken)
         {
             var result = await _credential.GetTokenAsync(
-                new TokenRequestContext(["api://AzureADTokenExchange/.default"], null),
+                new TokenRequestContext([_tokenExchangeUrl], null),
```

Comment:
Here the credential requires a scope. Scope = appIDUri+"./default" for app only tokens:

```suggestion
                new TokenRequestContext([_tokenExchangeUrl+"./default"], null),
```

Reply:
Applied 😄